subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -40,6 +40,55 @@ The previous step will help you identify potential problems by issuing deprecati
 
 === Potential breaking changes
 
+[[legacy_attribute_snapshotting]]
+==== ArtifactView and ArtifactCollection variant resolution behavior is more correct
+
+Prior to Gradle 8.3, `Configuration.getIncoming().artifactView()` and `Configuration.getIncoming().getArtifacts()` captured immutable copies of the `Configuration`'s attributes upon invocation. Consequently, if the `Configuration`'s attributes were changed after these methods were called, the attributes used for resolution by these methods could unexpectedly differ from the final `Configuration` attributes.
+
+Given that some attributes are lazily added to `Configuration`s during resolution, the behavior of these methods could have differed based on the order that these methods were called.
+
+Consider the following scenario where `producer` is a sibling `java-library` project to the build below:
+
+```groovy
+    plugins {
+        id 'java-library'
+    }
+
+    dependencies {
+        api project(':producer')
+    }
+
+    def incomingFiles = configurations.compileClasspath.incoming.files
+    def artifactViewFiles = configurations.compileClasspath.incoming.artifactView { }.files
+
+    println 'Incoming Files:'
+    incomingFiles.each { println 'Name: ' + it.name }
+
+    println 'Artifact View Files:'
+    artifactViewFiles.each { println 'Name: ' + it.name }
+
+    assert incomingFiles*.name == artifactViewFiles*.name // FAILS!
+```
+
+Running `gradle help` shows the following logger output that demonstrates the problem:
+
+```
+Incoming Files:
+Name: main
+Artifact View Files:
+Name: producer.jar
+```
```

Comment:
Like above, we shouldn't be resolving configurations at configuration time or suggesting that people use `gradle help`.   Can we simplify this example to not use java-library and show an example of where this didn't work before and is now fixed?